FF93 First changes for digest SHA256 support (#8869)
mdn/content: files/en-us/web/http/headers/www-authenticate/index.md

---
{{HTTPSidebar}}

The HTTP **`WWW-Authenticate`** response header defines the [HTTP authentication](/en-US/docs/Web/HTTP/Authentication) methods ("challenges") that might be used to gain access to a specific resource.

> **Note:** This header is part of the [General HTTP authentication framework](/en-US/docs/Web/HTTP/Authentication#the_general_http_authentication_framework), which can be used with a number of [authentication schemes](/en-US/docs/Web/HTTP/Authentication#authentication_schemes) .
> Each "challenge" lists a scheme supported by the server and additional parameters that are defined for that scheme type. 

> A server using [HTTP authentication](/en-US/docs/Web/HTTP/Authentication) will respond with {{HTTPStatus("401")}} `Unauthorized` response to a request for a protected resource.

Reviewer: ```suggestion
A server using [HTTP authentication](/en-US/docs/Web/HTTP/Authentication) will respond with a {{HTTPStatus("401")}} `Unauthorized` response to a request for a protected resource.
```

This response must include at least one `WWW-Authenticate` header and at least one {{Glossary("challenge")}}, to indicate what authentication schemes can be used to access the resource (and any additional data that each particular scheme needs).

> Multiple challenges are allowed in the one `WWW-Authenticate` header, and multiple `WWW-Authenticate` headers are allowed in the one response.

Reviewer: ```suggestion
Multiple challenges are allowed in one `WWW-Authenticate` header, and multiple `WWW-Authenticate` headers are allowed in one response.
```
…or ignore this suggestion if you in fact have *the* in there intentionally (which I can see is possible)

A server may also include the `WWW-Authenticate` header in other response messages to indicate that supplying credentials might affect the response.

> After recieving `WWW-Authenticate` a client will typically prompt the user for credentials, and then re-request the resource.

Reviewer: ```suggestion
After receiving the `WWW-Authenticate` header, a client will typically prompt the user for credentials, and then re-request the resource.
```

## Syntax

At least one challenge must be specified.

> Multiple challenges may be specified commas-separated in a single header, or in individual headers:

Reviewer: ```suggestion
Multiple challenges may be specified, comma-separated, in a single header, or in individual headers:
```

- **realm=**\<realm> {{optional_inline}}
  - : A string describing a protected area.
    A realm allows a server to partition up the areas it protects (if supported by a scheme that allows such partitioning).
    If no realm is specified, clients often display a formatted hostname instead.
- `<token68>` {{optional_inline}}
  - : A token that may be useful for some schemes. The token allows the 66 unreserved URI characters plus a few others.
    According to the specification, it can hold a base64, base64url, base32, or base16 (hex) encoding, with or without padding, but excluding whitespace.

Authorization parameter keys that are specific to each [authentication scheme](/en-US/docs/Web/HTTP/Authentication#authentication_schemes).
These are listed below for some schemes.
For others you may need to check the specifications:

### Basic

- **`<realm>`** {{optional_inline}}
  - : As above.
- **`charset="UTF-8"`** {{optional_inline}}
  - : Tells the client the server's preferred encoding scheme when submitting a username and password.

>       The only allowed value is the case insensitive string "UTF-8".

Reviewer: ```suggestion
      The only allowed value is the case-insensitive string "UTF-8".
```
